src/test/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4KitTest.kt

```diff
@@ -121,6 +144,422 @@ class GoogleAnalyticsFirebaseGA4KitTest {
+            .document("parental_consent_agreement_v2")
+            .location("17 Cherry Tree Lan 3")
+            .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+            .build()
+        val ccpaConsent =
+            CCPAConsent.builder(true) // true represents a "data sale opt-out", false represents the user declining a "data sale opt-out"
+                .document("ccpa_consent_agreement_v3")
+                .location("17 Cherry Tree Lane")
+                .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+                .build()
+        val state = ConsentState.builder()
+            .addGDPRConsentState("Marketing", locationCollectionConsent)
+            .addGDPRConsentState("Performance", parentalConsent)
+            .setCCPAConsentState(ccpaConsent)
+            .build()
+        filteredMParticleUser = FilteredMParticleUser.getInstance(user, kitInstance)
+
+        kitInstance.onConsentStateUpdated(state, state, filteredMParticleUser)
+
+
+        TestCase.assertEquals(0, firebaseSdk.getConsentState().size)
+    }
+
+    @Test
+    fun onConsentStateUpdatedTest_No_consentMappingSDK() {
+        val map = HashMap<String, String>()
+        map["defaultAdStorageConsentSDK"] = "Granted"
+        map["defaultAnalyticsStorageConsentSDK"] = "Granted"
+        map["defaultAdUserDataConsentSDK"] = "Denied"
+        map["defaultAdPersonalizationConsentSDK"] = "Denied"
+
+        kitInstance.configuration =
+            KitConfiguration.createKitConfiguration(JSONObject().put("as", map.toMutableMap()))
+
+        val locationCollectionConsent = GDPRConsent.builder(true)
+            .document("Test consent")
+            .location("17 Cherry Tree Lane")
+            .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+            .build()
+
+        val parentalConsent = GDPRConsent.builder(true)
+            .document("parental_consent_agreement_v2")
+            .location("17 Cherry Tree Lan 3")
+            .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+            .build()
+        val ccpaConsent =
+            CCPAConsent.builder(true) // true represents a "data sale opt-out", false represents the user declining a "data sale opt-out"
+                .document("ccpa_consent_agreement_v3")
+                .location("17 Cherry Tree Lane")
+                .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+                .build()
+        val state = ConsentState.builder()
+            .addGDPRConsentState("Marketing", locationCollectionConsent)
+            .addGDPRConsentState("Performance", parentalConsent)
+            .setCCPAConsentState(ccpaConsent)
+            .build()
+        filteredMParticleUser = FilteredMParticleUser.getInstance(user, kitInstance)
+
+        kitInstance.onConsentStateUpdated(state, state, filteredMParticleUser)
+
+        val expectedConsentValue =
+            firebaseSdk.getConsentState().getKeyByValue("AD_STORAGE").toString()
```

Comment:
It looks like you're only testing the actual value of `AD_STORAGE` but not the other three default settings, but the next assertion seems to test if all 4 are in the payload. Can you write assertions for all 4 values individually?